<launch>
    <!-- Get launch params-->
    <arg name="sim" default="false" doc="true: activates simulation time, switches to simulation color settings and deactivates launching of an image provider" />
    <arg name="camera" default="true" doc="true: launches an image provider to get images from a camera (unless sim:=true)" />
    <arg name="basler" default="true" doc="true: launches the basler camera driver instead of the wolves image provider" />
    <arg name="dummyball" default="false" doc="true: does not start the ball detection to save resources" />
    <arg name="debug" default="false" doc="true: activates publishing of several debug images" />
    <arg name="use_game_settings" default="false" doc="true: loads additional game settings" />


    <!-- Set vision config-->
    <rosparam command="load" file="$(find bitbots_vision)/config/visionparams.yaml" />

    <!-- load special sim parameters-->
    <group if="$(arg sim)">
        <rosparam command="load" file="$(find bitbots_vision)/config/simparams.yaml" />
    </group>

    <!-- Set debug params-->
    <param name="vision_publish_debug_image" value="$(arg debug)" />
    <param name="vision_publish_HSV_mask_image" value="$(arg debug)" />
    <param name="vision_publish_field_mask_image" value="$(arg debug)" />
    <param name="ball_fcnn_publish_debug_img" value="$(arg debug)" />

    <!-- Start vision without ball detection-->
    <group if="$(arg dummyball)">
        <param name="neural_network_type" value="dummy" />
    </group>

    <!-- Start the vision-->
    <node pkg="bitbots_vision" type="vision.py" name="bitbots_vision" output="screen"></node>

    <!-- Load game settings -->
    <group if="$(arg use_game_settings)">
        <rosparam command="load" file="$(find bitbots_bringup)/config/game_settings.yaml" />
    </group>

    <!-- Start dynamic-color-space node -->
    <node pkg="bitbots_vision" type="dynamic_color_space.py" name="bitbots_dynamic_color_space" output="screen">
    </node>

    <!-- Start the camera only when necessary -->
    <group if="$(arg camera)">
        <group unless="$(arg sim)">
            <include unless="$(arg basler)" file="$(find wolves_image_provider)/launch/image_provider.launch" />
            <include if="$(arg basler)" file="$(find bitbots_bringup)/launch/basler_camera.launch" />
        </group>
    </group>

    <!-- Set use simulation time -->
    <group if="$(arg sim)">
        <param name="use_sim_time" value="true" />
    </group>

</launch>
